### PR TITLE
Update mypy to check entire package and fix fga module

### PR DESCRIPTION
**Pull request by tribble**
## Description
`mypy` wasn't configured to check the entire module. I've updated it to check the entire module, and I fixed the ensuing errros.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

**Comment by atainter** [comment]
Hey @tribble I just merged my PR with a fix to the FGA files

**Comment by atainter** [comment]
https://github.com/workos/workos-python/pull/328

File: workos/sso.py [hunk 1]
```diff
@@ -138,9 +138,7 @@ def get_profile(self, access_token: str) -> Profile:
         Returns:
             Profile
         """
-        response = self._http_client.request(
-            PROFILE_PATH, method=REQUEST_METHOD_GET, token=access_token
-        )
+        response = self._http_client.request(PROFILE_PATH, method=REQUEST_METHOD_GET)
```
Comment (mattgd) [inline workos/sso.py:141]: Realized this should actually be the following:
```
response = self._http_client.request(
    PROFILE_PATH,
    method=RequestMethod.GET,
    headers={**self._http_client.auth_header_from_token(access_token)},
)
```

**Comment by tribble** [comment]
Since @atainter has as fix ready, I'm closing this PR.

**Pull request closed without merging**